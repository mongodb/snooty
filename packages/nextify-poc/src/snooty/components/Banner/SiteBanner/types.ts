export interface SiteBannerContent {
  isEnabled: boolean;
  altText: string;
  imgPath?: string;
  tabletImgPath?: string;
  mobileImgPath?: string;
  bgColor?: string;
  text?: string;
  pillText?: string;
  url: string;
}
