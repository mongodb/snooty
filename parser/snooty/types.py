import enum
import hashlib
from dataclasses import dataclass
from pathlib import PurePath
from typing import Any, Callable, Dict, Set, List, Tuple, Optional, Union

SerializableType = Union[None, bool, str, int, float, Dict[str, Any], List[Any]]
EmbeddedRstParser = Callable[[str, int, bool], List[SerializableType]]


@dataclass
class Diagnostic:
    __slots__ = ('message', 'severity', 'start', 'end')

    class Level(enum.IntEnum):
        info = 0
        error = 1
        warning = 2

        @classmethod
        def from_docutils(cls, docutils_level: int) -> 'Diagnostic.Level':
            level = docutils_level - 1
            level = min(level, cls.warning)
            level = max(level, cls.info)
            return cls(level)

    severity: Level
    message: str
    start: Tuple[int, int]
    end: Tuple[int, int]

    @property
    def severity_string(self) -> str:
        return self.severity.name.title()

    @classmethod
    def create(cls, severity: Level, message: str,
               start: Union[int, Tuple[int, int]],
               end: Union[None, int, Tuple[int, int]] = None) -> 'Diagnostic':
        if isinstance(start, int):
            start_line, start_column = start, 0
        else:
            start_line, start_column = start

        if end is None:
            end_line, end_column = start_line, 1000
        elif isinstance(end, int):
            end_line, end_column = end, 1000
        else:
            end_line, end_column = end

        return cls(severity, message, (start_line, start_column), (end_line, end_column))

    @classmethod
    def warning(cls, message: str,
                start: Union[int, Tuple[int, int]],
                end: Union[None, int, Tuple[int, int]] = None) -> 'Diagnostic':
        return cls.create(cls.Level.warning, message, start, end)

    @classmethod
    def error(cls, message: str,
              start: Union[int, Tuple[int, int]],
              end: Union[None, int, Tuple[int, int]] = None) -> 'Diagnostic':
        return cls.create(cls.Level.error, message, start, end)


@dataclass
class StaticAsset:
    __slots__ = ('fileid', 'checksum', 'data')

    fileid: str
    checksum: str
    data: Optional[bytes]

    def __hash__(self) -> int:
        return hash(self.checksum)

    @classmethod
    def load(cls, fileid: str, path: PurePath) -> 'StaticAsset':
        with open(path, 'rb') as f:
            data = f.read()
        asset_hash = hashlib.blake2b(data, digest_size=32).hexdigest()
        return cls(fileid, asset_hash, data)


@dataclass
class Page:
    source_path: PurePath
    source: str
    ast: SerializableType
    diagnostics: List[Diagnostic]
    static_assets: Set[StaticAsset]
    category: Optional[str] = None

    def get_id(self) -> PurePath:
        if self.category:
            # Giza wrote out yaml file artifacts under a directory. e.g. steps-foo.yaml becomes
            # steps/foo.rst
            return self.source_path.parent.joinpath(
                PurePath(self.category),
                self.source_path.name.replace(f'{self.category}-', '', 1))
        return self.source_path
