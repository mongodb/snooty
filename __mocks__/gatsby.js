const gatsby = jest.requireActual('gatsby');

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  StaticQuery: jest.fn(),
  withPrefix: jest.fn().mockImplementation(str => str),
  useStaticQuery: jest.fn(),
};
