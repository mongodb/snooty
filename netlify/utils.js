import fs from 'fs';
import path from 'path';

/**
 * Parses build hook data from the expected temporary location
 * @returns {object | undefined}
 */
const parseBuildHookData = () => {
  // This file does not currently exist, but should be created on Netlify builds as part of `npm run build:netlify`.
  // The INCOMING_HOOK_BODY env var is not automatically passed along to functions, so we use a txt file to save it
  const relativeFilePath = '../build-hook.txt';
  const buildHookDataString = fs.readFileSync(path.resolve(__dirname, relativeFilePath), 'utf-8');
  if (!buildHookDataString) {
    console.log('No build hook data found.');
    return;
  }
  return JSON.parse(buildHookDataString);
};

export const constructResPayload = (event) => {
  const buildHookData = parseBuildHookData();
  const parsedEventBody = JSON.parse(event.body);
  // This is Netlify's default post-deployment payload. We include it with our custom data in case
  // we want to process any information
  const netlifyPayload = parsedEventBody.payload;
  return {
    netlifyPayload,
    ...buildHookData,
  };
};
