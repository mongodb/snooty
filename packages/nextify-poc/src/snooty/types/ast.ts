// Need to figure out which is node type and which is component type
type ComponentType =
  | Exclude<NodeType, 'block-quote' | 'directive' | 'directive_argument' | 'role' | 'target_identifier'>
  | 'banner'
  | 'blockquote'
  | 'button'
  | 'card'
  | 'card-group'
  | 'chapter'
  | 'chapters'
  | 'code'
  | 'collapsible'
  | 'community-driver'
  | 'io-code-block'
  | 'cond'
  | 'container'
  | 'cta'
  | 'cta-banner'
  | 'definitionList'
  | 'definitionListItem'
  | 'deprecated'
  | 'deprecated-version-selector'
  | 'describe'
  | 'emphasis'
  | 'extract'
  | 'field'
  | 'field_list'
  | 'figure'
  | 'footnote'
  | 'footnote_reference'
  | 'glossary'
  | 'guide-next'
  | 'heading'
  | 'hlist'
  | 'image'
  | 'include'
  | 'introduction'
  | 'kicker'
  | 'line'
  | 'line_block'
  | 'literal_block';

type NodeType =
  | 'admonition'
  | 'banner'
  | 'block-quote'
  | 'code'
  | 'root'
  | 'section'
  | 'heading'
  | 'reference'
  | 'directive'
  | 'directive_argument'
  | 'line'
  | 'line_block'
  | 'list'
  | 'list-table'
  | 'listItem'
  | 'text'
  | 'literal'
  | 'literalinclude'
  | 'definitionList'
  | 'definitionListItem'
  | 'emphasis'
  | 'method-selector'
  | 'only'
  | 'openapi-changelog'
  | 'paragraph'
  | 'procedure'
  | 'reference'
  | 'ref_role'
  | 'role'
  | 'release_specification'
  | 'rubric'
  | 'search-results'
  | 'section'
  | 'seealso'
  | 'sharedinclude'
  | 'strong'
  | 'substitution_reference'
  | 'tabs-selector'
  | 'time'
  | 'title_reference'
  | 'transition'
  | 'versionadded'
  | 'versionchanged'
  | 'tabs'
  | 'target'
  | 'target_identifier'
  | 'text'
  | 'wayfinding';

type RoleName =
  | 'abbr'
  | 'class'
  | 'command'
  | 'file'
  | 'guilabel'
  | 'icon'
  | 'highlight-blue'
  | 'highlight-green'
  | 'highlight-red'
  | 'highlight-yellow'
  | 'icon-fa5'
  | 'icon-fa5-brands'
  | 'icon-fa4'
  | 'icon-mms'
  | 'icon-charts'
  | 'icon-lg'
  | 'kbd'
  | 'red'
  | 'gold'
  | 'required'
  | 'sub'
  | 'subscript'
  | 'sup'
  | 'superscript'
  | 'link-new-tab';

type NodeName =
  | RoleName
  | AdmonitionName
  | 'list-table'
  | 'contents'
  | 'collapsible'
  | 'tabs'
  | 'tab'
  | 'facet'
  | 'toctree'
  | 'deprecated'
  | 'versionadded'
  | 'versionchanged';

interface Node {
  type: NodeType;
}

interface ParentNode extends Node {
  children: Node[];
}

interface Root extends ParentNode {
  options: Record<string, any>;
  fileid: string;
}

interface BlockQuoteNode extends ParentNode {
  type: 'block-quote';
}

interface HeadingNode extends ParentNode {
  type: 'heading';
  id: string;
}

interface ParagraphNode extends ParentNode {
  type: 'paragraph';
}

interface EmphasisNode extends ParentNode {
  type: 'emphasis';
}

interface StrongNode extends ParentNode {
  type: 'strong';
}

interface ReferenceNode extends ParentNode {
  type: 'reference';
  refuri: string;
}

interface Directive<TOptions = { [key: string]: string }> extends ParentNode {
  type: 'directive';
  name: NodeName;
  argument: Node[];
  domain?: string;
  options?: TOptions;
}

interface ListTableNode extends Directive {
  name: 'list-table';
  children: ListNode[];
  options?: {
    widths?: string;
    'header-rows'?: string;
  };
}

interface ListNode extends ParentNode {
  type: 'list';
  enumtype: 'unordered' | 'ordered';
  children: ListItemNode[];
}

interface ListItemNode extends ParentNode {
  type: 'listItem';
}

interface LiteralNode extends ParentNode {
  type: 'literal';
}

interface LineBlockNode extends ParentNode {
  type: 'line_block';
}

interface LineNode extends ParentNode {
  type: 'line';
}

interface TextNode extends Node {
  type: 'text';
  value: string;
}

interface DefinitionListNode extends ParentNode {
  type: 'definitionList';
}

interface DefinitionListItemNode extends ParentNode {
  type: 'definitionListItem';
  term: Node[];
}

interface CodeNode extends Node {
  type: 'code';
  lang: string;
  copyable: boolean;
  emphasize_lines: number[];
  value: string;
  linenos: boolean;
}

interface MethodNode extends ParentNode {
  type: 'target';
  name: 'method';
  domain: 'mongodb';
}

interface DirectiveArgumentNode extends ParentNode {
  type: 'directive_argument';
}

interface TargetIdentifierNode extends ParentNode {
  type: 'target_identifier';
  ids: string[];
}

interface valueNode extends Node {
  value: string;
}
interface titleReferenceNode {
  children: valueNode[];
}

type TwitterOptions = {
  creator: string;
  image: string;
  'image-alt': string;
  site: string;
  title: string;
};

interface TwitterNode extends Directive<TwitterOptions> {
  options: TwitterOptions;
}
interface VersionModifiedNode extends Directive {
  children: Node[];
}

type CollapsibleOptions = {
  heading?: string;
  sub_heading?: string;
  id?: string;
};

interface CollapsibleNode extends Directive<CollapsibleOptions> {
  type: 'directive';
  name: 'collapsible';
  options?: CollapsibleOptions;
}

interface ContentsOptions {
  local?: boolean;
  backlinks?: string;
  depth?: number;
  class?: string;
}

interface ContentsNode extends Directive<ContentsOptions> {
  type: 'directive';
  name: 'contents';
  options: ContentsOptions;
}

interface TabsNode extends Directive {
  type: 'directive';
  name: 'tabs';
}

interface TabOptions {
  tabid: string;
}

interface TabNode extends Directive<TabOptions> {
  type: 'directive';
  name: 'tab';
  options?: TabOptions;
}

interface FacetOptions {
  name: string;
  values: string;
}

interface FacetNode extends Directive<FacetOptions> {
  type: 'directive';
  name: 'facet';
  options?: FacetOptions;
}

type AdmonitionName = 'example' | 'note' | 'tip' | 'see' | 'seealso' | 'warning' | 'important';

interface AdmonitionNode extends Directive {
  type: 'directive';
  name: AdmonitionName;
}

interface TocTreeEntry {
  title?: string;
  slug?: string;
}

interface TocTreeOptions {
  osiris_parent: boolean;
}

interface TocTreeDirective extends Directive<TocTreeOptions> {
  type: 'directive';
  name: 'toctree';
  entries: TocTreeEntry[];
}

export type {
  ComponentType,
  NodeType,
  RoleName,
  NodeName,
  Node,
  ParentNode,
  Root,
  HeadingNode,
  ReferenceNode,
  TextNode,
  CodeNode,
  Directive,
  ListNode,
  ListTableNode,
  ListItemNode,
  ParagraphNode,
  LiteralNode,
  LineBlockNode,
  LineNode,
  DefinitionListNode,
  DefinitionListItemNode,
  MethodNode,
  TargetIdentifierNode,
  DirectiveArgumentNode,
  CollapsibleNode,
  CollapsibleOptions,
  ContentsNode,
  EmphasisNode,
  StrongNode,
  TabsNode,
  TabNode,
  FacetNode,
  AdmonitionNode,
  AdmonitionName,
  TocTreeEntry,
  TocTreeDirective,
  TwitterOptions,
  TwitterNode,
  VersionModifiedNode,
  titleReferenceNode,
  BlockQuoteNode,
};
