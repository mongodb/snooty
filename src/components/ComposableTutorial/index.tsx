import ComposableTutorial from './ComposableTutorial';

export { ComposableTutorial };
