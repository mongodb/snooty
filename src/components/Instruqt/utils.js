export const getLabURL = (embedValue) => {
  return `https://play.instruqt.com/embed${embedValue}`;
};
