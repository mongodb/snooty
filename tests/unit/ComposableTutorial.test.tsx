import React from 'react';
import { render } from '@testing-library/react';
import * as Gatsby from 'gatsby';
import * as ReachRouter from '@gatsbyjs/reach-router';
import * as BrowserStorage from '../../src/utils/browser-storage';
import ComposableTutorial from '../../src/components/ComposableTutorial/ComposableTutorial';

import { ComposableTutorialNode } from '../../src/types/ast';
import ComposableData from './data/Composable.test.json';

const renderComposable = () => render(<ComposableTutorial nodeData={ComposableData as ComposableTutorialNode} />);

// mock local storage, location (query params), and navigate fn
const mockedGetLocalValue = jest.spyOn(BrowserStorage, 'getLocalValue');
const mockedNavigate = jest.spyOn(Gatsby, 'navigate');
const mockedUseLocation = jest.spyOn(ReachRouter, 'useLocation') as jest.SpyInstance<Partial<Location>>;

jest.mock('../../src/context/chatbot-context', () => ({
  useChatbotModal: () => ({
    chatbotClicked: false,
    setChatbotClicked: jest.fn(),
    text: '',
    setText: jest.fn(),
  }),
  ChatbotProvider: ({ children }: { children: React.ReactNode }) => children,
}));

describe('Composable Tutorial component', () => {
  beforeEach(() => {
    mockedGetLocalValue.mockReset();
    mockedNavigate.mockReset();
  });

  it('navigates to default selections without local storage or url query params', async () => {
    renderComposable();
    expect(mockedNavigate).toHaveBeenCalledWith(
      '?interface=driver&language=nodejs&deployment-type=atlas&operator=queryString',
      { state: { preserveScroll: true } }
    );
  });

  it('reads from local storage if available', () => {
    // mock local storage
    // wait for render
    // check to see if navigate has been called with mocked local storage
    mockedGetLocalValue.mockReturnValueOnce({
      'deployment-type': 'self',
      interface: 'atlas-admin-api',
      operator: 'autocomplete',
    });
    renderComposable();
    expect(mockedNavigate).toHaveBeenCalledWith(
      '?deployment-type=self&interface=atlas-admin-api&operator=autocomplete',
      { state: { preserveScroll: true } }
    );
  });

  it('prioritizes query params over local storage if correct selections', () => {
    // mock local storage
    mockedGetLocalValue.mockReturnValueOnce({
      'deployment-type': 'self',
      interface: 'atlas-admin-api',
      operator: 'autocomplete',
    });
    // mock query params / location
    mockedUseLocation.mockReturnValueOnce({
      pathname: 'test-pathname',
      search: '?deployment-type=atlas&interface=driver&language=c&operator=queryString',
    });

    // wait for render
    const wrapper = renderComposable();
    expect(wrapper.asFragment()).toMatchSnapshot();
    // check to see content is visible
    expect(wrapper.findByDisplayValue('queryString')).toBeTruthy();
    expect(wrapper.findByDisplayValue('Language')).toBeTruthy();
  });

  it('removes incorrect query params', () => {
    // mock local value
    mockedGetLocalValue.mockReturnValueOnce({
      'deployment-type': 'self',
      interface: 'atlas-admin-api',
      operator: 'autocomplete',
    });
    // mock query params / location with unnecessary language selection (dependent on interface = drivers)
    mockedUseLocation.mockReturnValueOnce({
      pathname: 'test-pathname',
      search: '?deployment-type=atlas&interface=atlas-admin-api&language=c&operator=queryString',
    });

    renderComposable();

    // removed bad selection of language
    expect(mockedNavigate).toHaveBeenCalledWith(
      '?deployment-type=self&interface=atlas-admin-api&operator=autocomplete',
      { state: { preserveScroll: true } }
    );
  });
});
