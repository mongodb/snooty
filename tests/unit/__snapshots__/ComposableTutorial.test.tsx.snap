// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Composable Tutorial component prioritizes query params over local storage if correct selections 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: -webkit-sticky;
  position: sticky;
  top: 60px;
  background: var(--background-color-primary);
  -webkit-column-gap: 16px;
  column-gap: 16px;
  row-gap: 12px;
  justify-items: space-between;
  border-bottom: 1px solid #E8EDEB;
  padding-bottom: 24px;
  padding-top: 8px;
  z-index: 2;
}

@media only screen and (max-width: 767px) {
  .emotion-0 {
    -webkit-box-flex-wrap: wrap;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }
}

.emotion-1 {
  -webkit-flex: 1 1 200px;
  -ms-flex: 1 1 200px;
  flex: 1 1 200px;
  font-size: 13px;
  overflow: hidden;
  z-index: 800;
  max-width: 200px;
}

.emotion-1 label,
.emotion-1 button {
  font-size: inherit;
}

.emotion-1 label {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.emotion-1>label {
  color: var(--font-color-primary);
}

.emotion-1 button {
  margin-top: 3px;
}

.dark-theme .emotion-1>button {
  background-color: var(--gray-dark4);
  color: var(--gray-light3);
}

@media only screen and (max-width: 767px) {
  .emotion-1 {
    -webkit-flex: 0 1 calc(50% - 8px);
    -ms-flex: 0 1 calc(50% - 8px);
    flex: 0 1 calc(50% - 8px);
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2>label+button,
.emotion-2>p+button {
  margin-top: 3px;
}

.emotion-2>label {
  margin-bottom: 4px;
  text-transform: uppercase;
  color: var(--gray-dark1);
}

.emotion-2>button {
  margin-top: 0;
  height: 22px;
}

.emotion-3 {
  font-family: 'Euclid Circular A',Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: bold;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  font-size: 13px;
  line-height: 20px;
  pointer-events: none;
}

@media only screen and (max-width: 1024px) {
  .emotion-3 {
    font-size: 16px;
    line-height: 19px;
  }
}

.emotion-4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  padding: 0;
  margin: 0;
  background-color: transparent;
  border: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  z-index: 0;
  font-family: 'Euclid Circular A',Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  border-radius: 6px;
  background-color: #F9FBFA;
  border-color: #889397;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  font-weight: 500;
  height: 36px;
  font-weight: 400;
  background-color: #FFFFFF;
  width: 100%;
  font-size: 13px;
}

.emotion-4:focus {
  outline: none;
}

.emotion-4[disabled],
.emotion-4:disabled {
  pointer-events: none;
}

.emotion-4:active,
.emotion-4:focus,
.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4:hover,
.emotion-4:active {
  color: #001E2B;
  background-color: #FFFFFF;
  box-shadow: 0 0 0 3px #E8EDEB;
}

.emotion-4:focus {
  background-color: #FFFFFF;
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC;
}

.emotion-4>*:last-child {
  grid-template-columns: 1fr 16px;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-4>*:last-child>svg {
  justify-self: right;
  width: 16px;
  height: 16px;
}

.emotion-4>*:last-child>svg {
  color: #3D4F58;
}

.emotion-4>*:last-child {
  padding: 0 4px 0 12px;
}

.emotion-4:focus {
  box-shadow: 0 0 0 3px #0498EC;
  border-color: rgba(255, 255, 255, 0);
}

@media only screen and (max-width: 1024px) {
  .emotion-4 {
    height: 36px;
    font-size: 16px;
  }
}

.emotion-5 {
  overflow: hidden;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 5px;
}

.emotion-6 {
  display: grid;
  grid-auto-flow: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  pointer-events: none;
  position: relative;
  z-index: 0;
  padding: 0 11px;
  gap: 6px;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  gap: 4px;
  overflow: hidden;
}

.emotion-8 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 100%;
}

.emotion-9 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: #889397;
  height: 16px;
  width: 16px;
  justify-self: left;
}

.emotion-37 {
  margin-top: 24px;
}

.emotion-38>*:first-child:not(script):not(style) {
  margin-top: 24px;
}

.emotion-39 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  color: #001E2B;
  font-weight: 400;
  margin-bottom: 16px;
  color: var(--font-color-primary);
}

.emotion-39 strong,
.emotion-39 b {
  font-weight: 700;
}

<div
    class=""
  >
    <div
      class="emotion-0"
    >
      <div
        class="configurable-option emotion-1"
      >
        <div
          class="emotion-2"
        >
          <label
            class="emotion-3"
            for="select-50"
            id="select-49-label"
          >
            Interface
          </label>
          <button
            aria-controls="select-49-menu"
            aria-describedby="select-49-description"
            aria-disabled="false"
            aria-expanded="false"
            aria-invalid="false"
            aria-label="Select your Interface"
            aria-labelledby="select-49-label"
            class="emotion-4"
            data-leafygreen-ui="button"
            data-testid="leafygreen-ui-select-menubutton"
            id="select-50"
            type="button"
            value="driver"
          >
            <div
              class="emotion-5"
            />
            <div
              class="emotion-6"
            >
              <div
                class="emotion-7"
              >
                <div
                  class="emotion-8"
                >
                  Driver
                </div>
              </div>
              <svg
                alt=""
                aria-hidden="true"
                class="emotion-9"
                height="16"
                role="presentation"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M8.67903 10.7962C8.45271 11.0679 8.04729 11.0679 7.82097 10.7962L4.63962 6.97649C4.3213 6.59428 4.5824 6 5.06866 6L11.4313 6C11.9176 6 12.1787 6.59428 11.8604 6.97649L8.67903 10.7962Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
      <div
        class="configurable-option emotion-1"
      >
        <div
          class="emotion-2"
        >
          <label
            class="emotion-3"
            for="select-70"
            id="select-69-label"
          >
            Language
          </label>
          <button
            aria-controls="select-69-menu"
            aria-describedby="select-69-description"
            aria-disabled="false"
            aria-expanded="false"
            aria-invalid="false"
            aria-label="Select your Language"
            aria-labelledby="select-69-label"
            class="emotion-4"
            data-leafygreen-ui="button"
            data-testid="leafygreen-ui-select-menubutton"
            id="select-70"
            type="button"
            value="c"
          >
            <div
              class="emotion-5"
            />
            <div
              class="emotion-6"
            >
              <div
                class="emotion-7"
              >
                <div
                  class="emotion-8"
                >
                  C
                </div>
              </div>
              <svg
                alt=""
                aria-hidden="true"
                class="emotion-9"
                height="16"
                role="presentation"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M8.67903 10.7962C8.45271 11.0679 8.04729 11.0679 7.82097 10.7962L4.63962 6.97649C4.3213 6.59428 4.5824 6 5.06866 6L11.4313 6C11.9176 6 12.1787 6.59428 11.8604 6.97649L8.67903 10.7962Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
      <div
        class="configurable-option emotion-1"
      >
        <div
          class="emotion-2"
        >
          <label
            class="emotion-3"
            for="select-52"
            id="select-51-label"
          >
            Deployment Type
          </label>
          <button
            aria-controls="select-51-menu"
            aria-describedby="select-51-description"
            aria-disabled="false"
            aria-expanded="false"
            aria-invalid="false"
            aria-label="Select your Deployment Type"
            aria-labelledby="select-51-label"
            class="emotion-4"
            data-leafygreen-ui="button"
            data-testid="leafygreen-ui-select-menubutton"
            id="select-52"
            type="button"
            value="atlas"
          >
            <div
              class="emotion-5"
            />
            <div
              class="emotion-6"
            >
              <div
                class="emotion-7"
              >
                <div
                  class="emotion-8"
                >
                  Atlas (Cloud)
                </div>
              </div>
              <svg
                alt=""
                aria-hidden="true"
                class="emotion-9"
                height="16"
                role="presentation"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M8.67903 10.7962C8.45271 11.0679 8.04729 11.0679 7.82097 10.7962L4.63962 6.97649C4.3213 6.59428 4.5824 6 5.06866 6L11.4313 6C11.9176 6 12.1787 6.59428 11.8604 6.97649L8.67903 10.7962Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
      <div
        class="configurable-option emotion-1"
      >
        <div
          class="emotion-2"
        >
          <label
            class="emotion-3"
            for="select-54"
            id="select-53-label"
          >
            Operator
          </label>
          <button
            aria-controls="select-53-menu"
            aria-describedby="select-53-description"
            aria-disabled="false"
            aria-expanded="false"
            aria-invalid="false"
            aria-label="Select your Operator"
            aria-labelledby="select-53-label"
            class="emotion-4"
            data-leafygreen-ui="button"
            data-testid="leafygreen-ui-select-menubutton"
            id="select-54"
            type="button"
            value="queryString"
          >
            <div
              class="emotion-5"
            />
            <div
              class="emotion-6"
            >
              <div
                class="emotion-7"
              >
                <div
                  class="emotion-8"
                >
                  queryString
                </div>
              </div>
              <svg
                alt=""
                aria-hidden="true"
                class="emotion-9"
                height="16"
                role="presentation"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M8.67903 10.7962C8.45271 11.0679 8.04729 11.0679 7.82097 10.7962L4.63962 6.97649C4.3213 6.59428 4.5824 6 5.06866 6L11.4313 6C11.9176 6 12.1787 6.59428 11.8604 6.97649L8.67903 10.7962Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="emotion-37"
    >
      <div
        class="emotion-38"
      >
        <p
          class="emotion-39"
        >
          This content will only be shown when the selections are as follows:
* Interface - Drivers
* Language - C
* Deployment Type - atlas
* Operator (custom snooty.toml) - queryString
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
